variables.tf

```diff
@@ -2,8 +2,60 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "watsonx_data_name" {
+  type        = string
+  description = "The name of the watsonx.data instance."
+  default     = null
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where the watsonx data instance is created."
+}
+
+variable "region" {
+  type        = string
+  description = "The region to provision the watsonx data instance. "
+  default     = "us-south"
+  validation {
+    condition     = contains(["eu-de", "us-south", "eu-gb", "jp-tok", "au-syd"], var.region)
+    error_message = "You must specify 'eu-de', 'eu-gb', 'jp-tok', 'au-syd' or 'us-south' as the IBM Cloud region."
+  }
+}
+
+variable "existing_watsonx_data_instance_crn" {
+  type        = string
+  description = "The CRN of the an existing watsonx.data instance. If no value is passed, and new instance will be provisioned."
+  default     = null
+
+  validation {
```

Comment:
would it not make more sense to have this validation in the `watsonx_data_name` variable?